sdk/nodejs/x/automation/stack.ts

```diff
@@ -262,15 +321,46 @@ export class Stack {
         }
 
         args.push("--exec-kind", kind);
+
+        // Set up event log tailing
+        const logFile = createLogFile("preview");
+        args.push("--event-log", logFile);
+        const eventLog: EngineEvent[] = [];
+        const logPromise = this.readLines(logFile, (event) => {
+            eventLog.push(event);
+            if (opts?.onEvent) {
+                const onEvent = opts.onEvent;
+                onEvent(event);
+            }
+        });
+        const prePromise = this.runPulumiCmd(args);
+
         let preResult;
+        let tail;
         try {
-            preResult = await this.runPulumiCmd(args);
+            [preResult, tail] = await Promise.all([prePromise, logPromise]);
         } finally {
             onExit();
```

Comment:
I don't know either way - but do you have an intuition for whether/why `onExit` should be called before or after quitting the tailing?

Reply:
Hm, I don't *think* it matters either way. They are both cleanup tasks, independent of each other - this wasn't written in this order for any other reason than I had to pick one out of two possible options :)